distributed/cli/tests/test_dask_spec.py

```diff
@@ -1,33 +1,37 @@
+import random
 import sys
 
+import pytest
 import yaml
 
 from distributed import Client
 from distributed.utils_test import gen_cluster, gen_test, popen
 
 
+@pytest.mark.flaky(reruns=2)
 @gen_test(timeout=120)
 async def test_text():
+    port = random.randint(10000, 50000)
```

Comment:
Conflicts are highly unlikely here but if we wanted to be really safe a pytest fixture that handed out and reclaimed ports might be a nice to have in future.